pkg/sql/reassign_owned_by.go

```diff
@@ -13,15 +13,114 @@ package sql
 import (
 	"context"
 
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/dbdesc"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
 )
 
 // ReassignOwnedByNode represents a REASSIGN OWNED BY <name> TO <name> statement.
 // TODO(angelaw): to implement
-//type reassignOwnedByNode struct {
-//}
+type reassignOwnedByNode struct {
+	n *tree.ReassignOwnedBy
+}
 
 func (p *planner) ReassignOwnedBy(ctx context.Context, n *tree.ReassignOwnedBy) (planNode, error) {
+
+	// Check all roles in old roles exist
+	// And that current user has membership on old role (PostgresQL requirement)
+	// TODO(angelaw): Refactor the error checking out to authorization?
```

Comment:
Does this seem like checking that should be done in the `authorization.go` file?